src/Entity/SmsMessage.php

```diff
@@ -0,0 +1,411 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\sms\Entity\SmsMessage.
+ */
+
+namespace Drupal\sms\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+
+/**
+ * Defines the SMS message entity.
+ *
+ * The SMS message entity is used to maintain the message while it is queued to
+ * send to the gateway. After the message has been sent, the message may persist
+ * in the database as an archive record.
+ *
+ * @ContentEntityType(
+ *   id = "sms",
+ *   label = @Translation("SMS Message"),
+ *   base_table = "sms",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *   },
+ * )
+ */
+class SmsMessage extends ContentEntityBase implements SmsMessageInterface {
+
+  /**
+   * Following are implementors of plain SmsMessage interface.
+   *
+   * @see \Drupal\sms\Entity\SmsMessageInterface
+   */
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRecipients() {
+    $recipients = [];
+    foreach ($this->get('recipient') as $recipient) {
+      $recipients[] = $recipient->value;
+    }
+    return $recipients;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addRecipient($recipient) {
+    $this->recipient->appendItem($recipient);
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addRecipients(array $recipients) {
+    foreach ($recipients as $recipient) {
+      $this->addRecipient($recipient);
+    }
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function removeRecipient($recipient) {
+    $this->recipient->filter(function ($item) use ($recipient) {
+      return ($item->value != $recipient);
+    });
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function removeRecipients(array $recipients) {
+    $this->recipient->filter(function ($item) use ($recipients) {
+      return !in_array($item->value, $recipients);
+    });
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOptions() {
+    return ($first = $this->get('options')->first()) ? $first->getValue() : [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOption($name) {
+    $options = $this->getOptions();
+    return isset($options[$name]) ? $options[$name] : NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setOption($name, $value) {
+    $options = $this->getOptions();
+    $options[$name] = $value;
+    $this->set('options', $options);
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function removeOption($name) {
+    $options = $this->getOptions();
+    unset($options[$name]);
+    $this->set('options', $options);
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSender() {
```

Comment:
hmm i was considering changing the FALSE value to NULL instead of empty string. Letting whatever caller to `getSender` decide what should be used (such as the raw phone number).